File: app/src/main/java/com/billpayments/features/payments/ui/HistoryScreenData.kt
package com.billpayments.features.payments.ui

import androidx.compose.runtime.Immutable
import com.billpayments.features.payments.data.PaymentStatus
import java.time.Instant

@Immutable
data class PaymentListItemData(
    val id: String,
    val billerName: String,
    val account: String,
    val amountPaise: Long,
    val paidAt: Instant,
    val status: PaymentStatus,
)

@Immutable
sealed interface HistoryScreenData {
    data object Empty : HistoryScreenData
    data class Loaded(val items: List<PaymentListItemData>) : HistoryScreenData
}

File: BillPayments/Features/Payments/UI/HistoryScreenData.swift
import Foundation

struct PaymentListItemData: Equatable, Identifiable {
    let id: String
    let billerName: String
    let account: String
    let amountPaise: Int64
    let paidAt: Date
    let status: PaymentStatus
}

enum HistoryScreenData: Equatable {
    case empty
    case loaded(items: [PaymentListItemData])
}
